apps/main/src/dex/lib/pools.ts

```diff
@@ -0,0 +1,145 @@
+
+export const hasNoWrapped = (pool: Pool) => pool?.isPlain || pool?.isFake
+
+const getPoolData = (p: Pool, network: NetworkConfig, storedPoolData: PoolData | undefined) => {
+  const isWrappedOnly = network.poolIsWrappedOnly[p.id]
+  const tokensWrapped = p.wrappedCoins.map((token, idx) => token || shortenAddress(p.wrappedCoinAddresses[idx])!)
+  const tokens = isWrappedOnly
+    ? tokensWrapped
+    : p.underlyingCoins.map((token, idx) => token || shortenAddress(p.underlyingCoinAddresses[idx])!)
+  const tokensLowercase = tokens.map((c) => c.toLowerCase())
+  const tokensAll = isWrappedOnly ? tokensWrapped : [...tokens, ...tokensWrapped]
+  const tokenAddresses = isWrappedOnly ? p.wrappedCoinAddresses : p.underlyingCoinAddresses
+  const tokenAddressesAll = isWrappedOnly
+    ? p.wrappedCoinAddresses
+    : [...p.underlyingCoinAddresses, ...p.wrappedCoinAddresses]
+  const tokenDecimalsAll = isWrappedOnly ? p.wrappedDecimals : [...p.underlyingDecimals, ...p.wrappedDecimals]
+  const tokensCountBy = countBy(tokens)
+
+  const poolData: PoolData = {
+    pool: p,
+    chainId: network.chainId,
+    curvefiUrl: '',
+
+    // stats
+    currenciesReserves: null,
+    parameters: storedPoolData?.parameters
+      ? storedPoolData.parameters
+      : {
+          A: '',
+          adminFee: '',
+          fee: '',
+          future_A: undefined,
+          future_A_time: undefined,
+          gamma: undefined,
+          initial_A: undefined,
+          initial_A_time: undefined,
+          lpTokenSupply: '',
+          virtualPrice: '',
+        },
+    hasVyperVulnerability: p.hasVyperVulnerability(),
+    hasWrapped: isWrappedOnly ?? !hasNoWrapped(p),
+    isWrapped: isWrappedOnly ?? false,
+    tokenAddressesAll,
+    tokenDecimalsAll,
+    tokenAddresses,
+    tokens,
+    tokensCountBy,
+    tokensAll,
+    tokensLowercase,
+    failedFetching24hOldVprice: false,
+    gauge: {
+      status: null,
+      isKilled: null,
+    },
+  }
+
+  return poolData
+}
+
+export async function getPools(
+  curve: CurveApi,
+  poolList: string[],
+  network: NetworkConfig,
+  poolsMapper: PoolDataMapper = {},
+  failedFetching24hOldVprice?: { [p: string]: boolean } | null,
+) {
+  const { getPool } = curve
+  const { orgUIPath } = network
+
+  const resp = poolList.reduce(
+    (prev, poolId) => {
+      const pool = getPool(poolId)
+      const poolData = getPoolData(pool, network, poolsMapper[poolId])
+
+      poolData.failedFetching24hOldVprice = failedFetching24hOldVprice?.[pool.address] ?? false
+      poolData.curvefiUrl = getCurvefiUrl(poolId, orgUIPath)
+
+      prev.poolsMapper[poolId] = poolData
+
+      prev.poolsMapperCache[poolId] = pick(poolData, [
+        'hasWrapped',
+        'gauge',
+        'tokens',
+        'tokensCountBy',
+        'tokensAll',
+        'tokensLowercase',
+        'tokenAddresses',
+        'tokenAddressesAll',
+        'pool.id',
+        'pool.name',
+        'pool.address',
+        'pool.gauge.address',
```

Comment:
This function is pretty much just moved elsewhere. Note though that I changed this line from `'pool.gauge'` - the whole object contains a class that cannot be serialized